src/Query/Builder.php

```diff
@@ -1660,7 +1663,10 @@ private function aliasIdForResult(array|object $values): array|object
             }
 
             foreach ($values as $key => $value) {
-                if (is_array($value) || is_object($value)) {
+                if ($value instanceof UTCDateTime) {
+                    $values[$key] = Date::instance($value->toDateTime())
+                        ->setTimezone(new DateTimeZone(date_default_timezone_get()));
+                } elseif (is_array($value) || $value instanceof stdClass) {
```

Comment:
This changes `is_object()` to a stdClass check. Was that intentional?

I'm not familiar with how `aliasIdForResult()` is used, but this looks like it may be an optimization to avoid unnecessary recursion. `aliasIdForResult()` only handled array and stdClass instances and returned `$values` as-is otherwise.

Reply:
Yes, but this method may accept any class defined in the typeMap. Not something supported for now as the typeMap cannot be modified. Tracked in [PHPORM-233](https://jira.mongodb.org/browse/PHPORM-233)